include/slam_toolbox/visualization_utils.hpp

```diff
@@ -20,24 +20,28 @@
 #define SLAM_TOOLBOX__VISUALIZATION_UTILS_HPP_
 
 #include <string>
+#include <memory>
 
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
 #include "visualization_msgs/msg/marker.hpp"
 #include "visualization_msgs/msg/interactive_marker.hpp"
 #include "visualization_msgs/msg/interactive_marker_feedback.hpp"
 
 namespace vis_utils
 {
 
+template<class ClockHandler>
 inline visualization_msgs::msg::Marker toMarker(
   const std::string & frame,
   const std::string & ns,
   const double & scale,
-  rclcpp::Node::SharedPtr node)
+  std::shared_ptr<ClockHandler> handler)
```

Comment:
Passing the node here + the one below feels more correct. Unless you want to change the API to just passing the time itself instead of the node or clock.

Reply:
https://github.com/SteveMacenski/slam_toolbox/pull/561/commits/37648d73307b3480a422282e1e4686dee9d4f6b8 ( Same commit as <https://github.com/SteveMacenski/slam_toolbox/pull/561#discussion_r1055518903> )

Changed to the node